src/utils/util.ts

```diff
@@ -72,6 +72,41 @@ ${docs.tags
   .join('\n')}`.trim();
 }
 
+/**
+ * Adds a doc block to a string
+ * @param str the string to add a doc block to
+ * @param docs the compiled JS docs
+ * @param indentation number of spaces to indent the block with
+ * @returns the doc block
+ */
+export function addDockBlock(str: string, docs?: d.CompilerJsDoc, indentation: number = 0): string {
+  if (!docs) {
+    return str;
+  }
+
+  return [getDocBlock(docs, indentation), str].filter(Boolean).join(`\n`);
+}
+
+function getDocBlock(docs: d.CompilerJsDoc, indentation: number = 0): string {
+  const textDocs = getDocBlockLines(docs);
+  if (!textDocs.filter(Boolean).length) {
+    return '';
+  }
+
+  const spaces = new Array(indentation + 1).join(' ');
+
+  return [spaces + '/**', ...textDocs.map((line) => spaces + ` * ${line}`), spaces + ' */'].join(`\n`);
+}
+
+function getDocBlockLines(docs: d.CompilerJsDoc | undefined | null): string[] {
```

Comment:
I don't think it can be `undefined | null` actually, there is a check for that in the `addDocBlock` function and it will not be invoked from anywhere else. I updated the types to make it more clear

a951d30